crates/primitives/src/bytecode/eof.rs

```diff
@@ -182,8 +182,7 @@ impl fmt::Display for EofDecodeError {
     }
 }
 
-#[cfg(feature = "std")]
-impl std::error::Error for EofDecodeError {}
+impl core::error::Error for EofDecodeError {}
```

Comment:
FYI this increases MSRV to 1.81 https://blog.rust-lang.org/2024/09/05/Rust-1.81.0.html#coreerrorerror